### ISSUE TITLE
test: auto generate test certificate

### ISSUE THREAD

**Issue by climba03003** [fastify/fastify-jwt#145]
## 🚀 Feature Proposal

Move the current static test certification to auto generate before the test start.
https://github.com/fastify/fastify-jwt/tree/master/certs

## Motivation

It can reduce the `false-positive` warning from security scanner.

**Comment by jsumners** [comment]
This code generates a compatible certificate, jwt, and jwks:

```js
'use strict';

const forge = require('node-forge');
const jsonwebtoken = require('jsonwebtoken');
const { pem2jwk } = require('pem-jwk');

module.exports = { generateJWT };

/**
 * Generates a fake JSON Web Token (JWT) and JSON Web Token Key Set (JWKS).
 *
 * @example
 * const { jwt, jwks } = generateJWT({
 *   payload: { foo: 'foo' }
 * })
 * // jwt = 'part1.part2.part3'
 * // jwks = [{ use: 'sig', kid: 'test-key', ... }]
 *
 * @param {object} [input]
 * @param {object} [input.payload={}] An object that will be me merged with the
 * base JWT object (which has things like `iss` and `scope`). So, for example,
 * if you want the resulting JWT to have a `https://example` property you
 * would supply `payload = { 'https://example': { key: 'value' } }`.
 * @param {string} [input.keyid='test-key'] The name for the token signing key.
 * @param {string} [input.accountId='1234'] An identifier for the account
 * the token will represent. This will be available in the `sub` property
 * and prefixed by `auth0|`, e.g. `sub = 'auth0|1234'`.
 * @param {integer|string} [input.expiresIn=60] The time, in seconds, the
 * JWT will be valid for. This value can also be a string supported by the
 * `ms` module, e.g '30m' for thirty minutes.
 *
 * @returns {object} An object with properties `jwt` and `jwks`. The `jwt`
 * property can be used to build an authorization header, e.g.
 * `authorization = 'Bearer ' + jwt`. The `jwks` value can be used to respond
 * for requests to a JWKS URL via a `nock` intercept.
 */
function generateJWT({
  payload = {},
  keyid = 'test-key',
  accountId = '1234',
  expiresIn = 60
} = {}) {
  const jwtInput = Object.assign(
    {
      iss: 'https://example.auth0.com/',
      sub: `auth0|${accountId}`,
      aud: ['https://example.com', 'https://example.auth0.com/userinfo'],
      scope: 'openid profile email address phone',
      gty: 'password'
    },
    payload
  );
  const rsaKeyPair = forge.pki.rsa.generateKeyPair({ bits: 2048, e: 0x10001 });
  const privatePEM = forge.pki.privateKeyToPem(rsaKeyPair.privateKey);
  const publicPEM = forge.pki.publicKeyToPem(rsaKeyPair.publicKey);
  const publicJWK = pem2jwk(publicPEM);

  const jwks = {
    keys: [{ use: 'sig', kid: keyid, ...publicJWK }]
  };
  const jwt = jsonwebtoken.sign(jwtInput, privatePEM, {
    algorithm: 'RS256',
    expiresIn,
    keyid
  });

  return { jwt, jwks };
}
```

**Comment by simone-sanfratello** [comment]
working on that

**Comment by climba03003** [comment]
Resolved #146
